src/main/java/database/service/JsonDatabaseService.java

```diff
@@ -0,0 +1,272 @@
+        if (isEntityRemoved) {
+            serializeEntities(entities, databasePath);
+        }
+        return isEntityRemoved;
+    }
+
+    @Override
+    public void removeAllRecordsFromTable(Class<? extends BaseEntity> entityClass) {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+        try {
+            Files.writeString(databasePath, EMPTY_BRACKETS_TO_JSON);
+        } catch (IOException e) {
+            logger.error("Unable to serialize data at " + databasePath.toAbsolutePath());
+            throw new SerializeDatabaseException("Unable to serialize data.");
+        }
+    }
+
+    @Override
+    public <T extends BaseEntity, I> T getById(Class<? extends BaseEntity> entityClass, I id) throws IOException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        List<T> entities = deserializeEntities(entityClass, Files.readString(databasePath));
+
+        return entities.stream()
+                .filter(e -> id.equals(e.getId()))
+                .findFirst()
+                .orElse(null);
+    }
+
+    @Override
+    public <T extends BaseEntity> Iterable<T> getAllRecordsFromTable(Class<? extends BaseEntity> entityClass)
+            throws IOException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        return deserializeEntities(entityClass, Files.readString(databasePath));
+    }
+
+    @Override
+    public <T extends BaseEntity, V> Iterable<T> getByFilters(Class<? extends BaseEntity> entityClass,
+                                                              Map<String, V> filters) throws IOException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+        Field[] fields = entityClass.getDeclaredFields();
+        Validator.validateDatabaseFilters(fields, filters);
+
+        List<T> entities = deserializeEntities(entityClass, Files.readString(databasePath));
+
+        return entities.stream()
+                .filter(entity -> {
+                    for (Map.Entry<String, V> filter : filters.entrySet()) {
+                        String fieldName = filter.getKey();
+                        V expectedValue = filter.getValue();
+                        String getterName = getGetterName(entityClass, fieldName);
+
+                        try {
+                            Method getterMethod = entityClass.getMethod(getterName);
+                            Object actualValue = getterMethod.invoke(entity);
+
+                            if (!expectedValue.equals(actualValue)) {
+                                return false;
+                            }
+                        } catch (IllegalAccessException e) {
+                            logger.error("Unable to access property " + fieldName);
+                            e.printStackTrace();
+                            return false;
+                        } catch (InvocationTargetException | NoSuchMethodException e) {
```

Comment:
Please combine all 3 exceptions into one `catch`, I don't see a reason to separately handle them.